cmd/gctcli/commands.go

```diff
@@ -2270,3 +2272,372 @@ func disableExchangePair(c *cli.Context) error {
+
+	if c.IsSet("exchange") {
+		exchangeName = c.String("exchange")
+	} else {
+		exchangeName = c.Args().First()
+	}
+
+	if c.IsSet("pair") {
+		pair = c.String("pair")
+	} else {
+		pair = c.Args().Get(1)
+	}
+
+	if !validPair(pair) {
+		return errInvalidPair
+	}
+
+	if c.IsSet("asset") {
+		assetType = c.String("asset")
+	} else {
+		assetType = c.Args().Get(2)
+	}
+
+	p := currency.NewPairDelimiter(pair, pairDelimiter)
+
+	client := gctrpc.NewGoCryptoTraderClient(conn)
+	result, err := client.GetOrderbookStream(context.Background(),
+		&gctrpc.GetOrderbookStreamRequest{
+			Exchange: exchangeName,
+			Pair: &gctrpc.CurrencyPair{
+				Base:      p.Base.String(),
+				Quote:     p.Quote.String(),
+				Delimiter: p.Delimiter,
+			},
+			AssetType: assetType,
+		},
+	)
+
+	if err != nil {
+		return err
+	}
+
+	for {
+		resp, err := result.Recv()
+		if err != nil {
+			return err
+		}
+
+		clear := exec.Command("clear")
+		clear.Stdout = os.Stdout
+
+		err = clear.Run()
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("Orderbook stream for %s %s:\n\n", exchangeName,
+			resp.Pair.String())
+		fmt.Println("\t\tBids\t\t\t\tAsks")
+		fmt.Println()
+
+		bidLen := len(resp.Bids) - 1
+		askLen := len(resp.Asks) - 1
+
+		var maxLen int
+		if bidLen >= askLen {
+			maxLen = bidLen
+		} else {
+			maxLen = askLen
+		}
+
+		for i := 0; i < maxLen; i++ {
+			var bidAmount, bidPrice float64
+			if i <= bidLen {
+				bidAmount = resp.Bids[i].Amount
+				bidPrice = resp.Bids[i].Price
+			}
+
+			var askAmount, askPrice float64
+			if i <= askLen {
+				askAmount = resp.Asks[i].Amount
+				askPrice = resp.Asks[i].Price
+			}
+
+			fmt.Printf("%f %s @ %f %s\t\t%f %s @ %f %s\n",
+				bidAmount,
+				resp.Pair.Base,
+				bidPrice,
+				resp.Pair.Quote,
+				askAmount,
+				resp.Pair.Base,
+				askPrice,
+				resp.Pair.Quote)
+		}
+	}
+}
+
+var getExchangeOrderbookStreamCommand = cli.Command{
+	Name:      "getexchangeorderbookstream",
+	Usage:     "gets a stream for all orderbooks associated with an exchange",
+	ArgsUsage: "<exchange>",
+	Action:    getExchangeOrderbookStream,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "exchange",
+			Usage: "the exchange to get the orderbook from",
+		},
+	},
+}
+
+func getExchangeOrderbookStream(c *cli.Context) error {
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		cli.ShowCommandHelp(c, "getexchangeorderbookstream")
+		return nil
+	}
+
+	conn, err := setupClient()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	var exchangeName string
+
+	if c.IsSet("exchange") {
+		exchangeName = c.String("exchange")
+	} else {
+		exchangeName = c.Args().First()
+	}
+
+	client := gctrpc.NewGoCryptoTraderClient(conn)
+	result, err := client.GetExchangeOrderbookStream(context.Background(),
+		&gctrpc.GetExchangeOrderbookStreamRequest{
+			Exchange: exchangeName,
+		})
+
+	if err != nil {
+		return err
+	}
+
+	for {
+		resp, err := result.Recv()
+		if err != nil {
+			return err
+		}
+
+		clear := exec.Command("clear")
```

Comment:
This command wont execute on windows (at least in cmd)/some non-linux based operating systems 

I do like the screen clean before the output it helps keep it read able might be worth adding support for other operating systems instead of dropping completely